golem-cli/src/command.rs

```diff
@@ -32,3 +43,149 @@ impl ComponentRefSplit<OssContext> for ComponentUriArg {
         (self.uri, None)
     }
 }
+
+pub trait CliCommand<Ctx>: Subcommand {
+    fn run(self, ctx: Ctx) -> impl Future<Output = Result<GolemResult, GolemError>>;
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Zip<A: Subcommand, B: Subcommand> {
+    #[command(flatten)]
+    First(A),
+    #[command(flatten)]
+    Second(B),
```

Comment:
How about `Left` and `Right`, more aligned to what's being done here, that it keeps composing the right with what's in left that may be already flattened with first and second :)